### PR TITLE
Feature/t3 encbe 156 use typescript

### PR DESCRIPTION
**Pull request by NicoleNitschke**
use TypeScript instead of pure JS to prevent JavaScript Errors

### CODE REVIEW

File: Resources/Private/Frontend/JavaScript/Features/ContentTree.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@ function buildContentTree() {
   if (iframe == null) {
```
Comment (MarcinSzalomski) [inline Resources/Private/Frontend/JavaScript/Features/ContentTree.ts:20]: I would suggest using the strict type check with `===` or even check against a falsey value.

So either `if(iframe === null)` or `if(!iframe)`. However, the second option also checks against values like `undefined`, which can't really be delivered by the `querySelector` function

File: Resources/Private/Frontend/JavaScript/Features/ContentTree.ts [hunk 2]
```diff
@@ -49,30 +50,30 @@ function createContentTreeHTML() {
   contentTreeHeader.innerHTML = '<label class="content-tree__headline">Content tree</label><i class="fa fa-solid fa-angle-down content-tree__toggle"></i>';
   contentTreeHeader.onclick = initContentTreeToggle;
 
-  const contentTreeHeadline = contentArea.querySelector('.typo3-docheader-pagePath + strong') ? contentArea.querySelector('.typo3-docheader-pagePath + strong').innerHTML : (contentArea.querySelector('.t3js-title-inlineedit') ? contentArea.querySelector('.t3js-title-inlineedit').textContent : 'Content Tree');
+  const contentTreeHeadline = contentArea?.querySelector<HTMLElement>('.typo3-docheader-pagePath + strong') ? contentArea?.querySelector<HTMLElement>('.typo3-docheader-pagePath + strong')?.innerHTML : (contentArea?.querySelector<HTMLElement>('.t3js-title-inlineedit') ? contentArea?.querySelector<HTMLElement>('.t3js-title-inlineedit')?.textContent : 'Content Tree');
```
Comment (MarcinSzalomski) [inline Resources/Private/Frontend/JavaScript/Features/ContentTree.ts:53]: This line is pretty confusing. It includes nested checks. Would it be possible to use `if`-clauses for the presence checks?
Or even move it into a separate generator function?

The other option would be to move it to the next rows - just like you did in the line 114.

File: Resources/Private/Frontend/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    /* Interop Constraints */
+    // "isolatedModules": true,                          /* Ensure that each file can be safely transpiled without relying on other imports. */
+    // "verbatimModuleSyntax": true,                     /* Do not transform or elide any imports or exports not marked as type-only, ensuring they are written in the output file's format based on the 'module' setting. */
+    // "allowSyntheticDefaultImports": true,             /* Allow 'import x from y' when a module doesn't have a default export. */
+    "esModuleInterop": true,                             /* Emit additional JavaScript to ease support for importing CommonJS modules. This enables 'allowSyntheticDefaultImports' for type compatibility. */
+    // "preserveSymlinks": true,                         /* Disable resolving symlinks to their realpath. This correlates to the same flag in node. */
+    "forceConsistentCasingInFileNames": true,            /* Ensure that casing is correct in imports. */
+
+    /* Type Checking */
+    "strict": true,                                      /* Enable all strict type-checking options. */
```
Comment (MarcinSzalomski) [inline Resources/Private/Frontend/tsconfig.json:78]: If you want a stricter typescript checks, you could activate the following options. However, i would see that only as a suggestion. The change is not critical.

```
    "strict": true, 
    "noImplicitAny": false, 
    "strictNullChecks": true, 
    "strictFunctionTypes": true, 
    "strictBindCallApply": true,
    "strictPropertyInitialization": true,
    "noImplicitThis": true,
    "alwaysStrict": true,
```

**Review by MarcinSzalomski — CHANGES_REQUESTED** [review]

File: Resources/Private/Frontend/JavaScript/Features/ContentTree.ts (hunk 2 above)
Comment (NicoleNitschke) [reply Resources/Private/Frontend/JavaScript/Features/ContentTree.ts]: done :)

File: Resources/Private/Frontend/JavaScript/Features/ContentTree.ts [hunk 4]
```diff
@@ -148,7 +158,7 @@ function createNestedList(rootElement, classList, createdTreeLinks) {
  * furthermore, the content tree is rebuilt when the selected page changes
  */
 function watchContentIframe() {
-  document.querySelector('#typo3-contentIframe').addEventListener('load', (event) => {
+  document.querySelector<HTMLIFrameElement>('#typo3-contentIframe')?.addEventListener('load', (event) => {
     if (window.top !== window) {
```
[the remark this answers is not in the document]
Comment (NicoleNitschke) [reply Resources/Private/Frontend/JavaScript/Features/ContentTree.ts]: yes, it's improved now

File: Resources/Private/Frontend/package.json [hunk 5]
```diff
@@ -6,7 +6,8 @@
         "watch": "gulp watch",
         "watch-prod": "gulp watch --mode=production",
         "build": "gulp build",
-        "build-prod": "gulp build --mode=production"
+        "build-prod": "gulp build --mode=production",
+				"build-ts": "gulp ts:compile"
```
[the remark this answers is not in the document]
Comment (NicoleNitschke) [reply Resources/Private/Frontend/package.json]: I reformatted the file, the script is still usefull in some cases :)

File: Resources/Private/Frontend/tsconfig.json (hunk 3 above)
Comment (NicoleNitschke) [reply Resources/Private/Frontend/tsconfig.json]: We will do this later maybe! :)

File: Resources/Private/Frontend/JavaScript/Features/SaveSettingsListener.ts [hunk 6]
```diff
@@ -56,18 +56,21 @@ function updateFeatures(selectedPreset) {
   }
 }
 
-function setPreset(presetValue) {
-  document.querySelector('[name="data[enba-presets]"][value="' + presetValue + '"]').checked = true;
+function setPreset(presetValue : string) {
+  const checkbox = document.querySelector<HTMLInputElement>('[name="data[enba-presets]"][value="' + presetValue + '"]');
+  if (checkbox) {
```
[the remark this answers is not in the document]
Comment (NicoleNitschke) [reply Resources/Private/Frontend/JavaScript/Features/SaveSettingsListener.ts:61]: I think the origin version is better readbale and less confusing, though yours is smarter :)

File: Resources/Private/Frontend/JavaScript/Features/ContentTree.ts [hunk 7]
```diff
@@ -125,7 +135,7 @@ function createNestedList(rootElement, classList, createdTreeLinks) {
       }
 
       // Check if element is already in content tree, if yes, don't use it as content root
-      const currentElementURL = linkToContentElement.getAttribute('href');
+      const currentElementURL = linkToContentElement.getAttribute('href') || '';
```
[the remark this answers is not in the document]
Comment (NicoleNitschke) [reply Resources/Private/Frontend/JavaScript/Features/ContentTree.ts]: Done!

**Review by MarcinSzalomski — APPROVED** [review]

**Pull request merged**